server/configuration-validation.js

```diff
@@ -1,4 +1,171 @@
 const { isObject } = require('./utils')
+/**
+ * Validate and cleans the DB query settings.
+ * These will be used when fetching documents from the user's database.
+ *
+ *
+ * @param  {object} options
+ * @param  {string} options.collectionName
+ * @param  {object} options.configuration
+ * @param  {string[]} options.configuration.fields
+ * @param  {any} options.configuration.filters
+ * @param  {number} options.configuration.start
+ * @param  {number} options.configuration.limit
+ * @param  {any} options.configuration.sort
+ * @param  {any} options.configuration.populate
+ * @param  {any} options.configuration.publicationState
+ * @param  {any} options.configuration.locale
+ *
+ * @returns {object} - All validating functions
+ *
+ */
+function EntriesQuery({ configuration, collectionName }) {
+  const log = strapi.log // has to be inside a scope
+  const {
+    fields,
+    filters,
+    start,
+    limit,
+    sort,
+    populate,
+    publicationState,
+    locale,
+    ...unknownKeys
+  } = configuration
+
+  const options = {}
+
+  return {
+    validateFields() {
+      if (fields !== undefined && !Array.isArray(fields)) {
+        log.error(
+          `The "fields" option in "queryOptions" of "${collectionName}" should be an array of strings.`
+        )
+      } else if (fields !== undefined) {
```

Comment:
Why not just and `else` ?

Reply:
because it is `undefined` it should not enter neither the `if` nor the `else if`.